tests.py

```diff
@@ -1491,7 +1492,36 @@ def test_nickname_and_last_name_with_title(self):
         self.m(hn.last, "Edmonds", hn)
         self.m(hn.nickname, "Rick", hn)
 
+    def test_append_nickname(self):
+        hn = HumanName()
+        new_rgx = re.compile(r'(?!\w)\(_open(\w[^)]*?)\):close(?!\w)', re.UNICODE)
+        hn._nickname_regexes.append(new_rgx)
+        self.assertEqual(hn._nickname_regexes[-1], new_rgx)
+        hn.full_name = r"Benjamin (_openBen):close Franklin"
+        self.m(hn.first, "Benjamin", hn)
+        self.m(hn.middle, ":close", hn)
+        self.m(hn.last, "Franklin", hn)
+        self.m(hn.nickname, "_openBen", hn)
 
+    def test_prepend_nickname(self):
+        hn = HumanName()
+        new_rgx = re.compile(r'(?!\w)\(_open(\w[^)]*?)\):close(?!\w)', re.UNICODE)
+        hn._nickname_regexes.insert(0, new_rgx)
+        self.assertEqual(hn._nickname_regexes[0], new_rgx)
+        hn.full_name = r"Benjamin (_openBen):close Franklin"
+        self.m(hn.first, "Benjamin", hn)
+        self.m(hn.middle, "", hn)
+        self.m(hn.last, "Franklin", hn)
+        self.m(hn.nickname, "Ben", hn)
+
+    def test_multiple_nicknames(self):
+        hn = HumanName('Chief Justice John (JR), "No Glove, No Love" Glover Roberts, Jr.')
+        self.m(hn.title, 'Chief Justice', hn)
+        self.m(hn.first, "John", hn)
+        self.m(hn.middle, "Glover", hn)
+        self.m(hn.last, "Roberts", hn)
+        self.m(hn.suffix, "Jr.", hn)
+        self.m(hn.nickname, '"JR", "No Glove, No Love"', hn)
```

Comment:
Since there is a list variable specifically for nickname regexes, adding one with a tag/label isn't required.  Those would be added at the source code level, not at run time.

Reply:
Sure, but we should have a test that adds them at the source code level then, just so that code gets run and we know if we ever break that feature, since it will be supported now.

Reply:
Tests are for **run-time actions**.  The only way we would know that someone had made a change to the source code was if we kept a secret copy of the source code to be used as a comparison.

I have a test that adds a nickname regex twice.  The first time, it appends a pattern, which gives undesired results, since the existing/original nickname patterns would match first.  It then prepends the same pattern to the list and it results in the desired result.

Reply:
Looking more closely, I think the answer is that your test is already using the label so it would fail if that feature wasn't supported, but that it's more of a feature of regex than the parser anyway so it doesn't make sense for us to test it.